astropy/units/quantity.py

```diff
@@ -39,6 +40,15 @@
 _UNIT_NOT_INITIALISED = "(Unit not initialised)"
 _UFUNCS_FILTER_WARNINGS = {np.arcsin, np.arccos, np.arccosh, np.arctanh}
 
+def quantity_allclose(a, b, rtol=1.e-5, atol=None, **kwargs):
+    """
+    Returns True if two arrays are element-wise equal within a tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
+    :func:`numpy.allclose`.
+    """
+    return np.allclose(*_unquantify_allclose_arguments(a, b, rtol, atol),
```

Comment:
@mhvk , should `_unquantify_allclose_arguments` be moved over too? (Since this is a "hidden" function, I don't think deprecation would be necessary if we decide to move it.)

Reply:
Surely without moving that over `astropy.units.quantity` now has an import dependency on pytest?!

Reply:
Yes, `_unquantity_allclose_arguments` definitely needs to be moved here.

Reply:
And the name `_unquantity_allclose_arguments ` shall remain the same or changed like `allclose` ?

Reply:
I think the name is fine as is -- and it is a private function anyway, so we can change it later if needed.